packages/slice-machine/lib/models/common/http/DefaultClient.ts

```diff
@@ -129,21 +130,21 @@ export default class DefaultClient {
     auth: string
   ): Promise<Error | UserProfile> {
     try {
-      const result = await fetch(`${createApiUrl(base, UserService)}profile`, {
-        method: "GET",
-        headers: {
-          Authorization: `Bearer ${auth}`,
-        },
-      });
-
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      const jsResult = await result.json();
+      const result = await axios.get<UserProfile>(
```

Comment:
The point of using trying to decode the user profile given at line 145 is that until that point we are not sure what's the shape of the data.

You can use axios, I would understand that you prefer it to fetch but you can't cast with it to a UserProfile.
That's the job of the decode later on, to validate this data.


I would type it to unknown (common type used by io-ts before validation.
```suggestion
      const result = await axios.get<unknown>(
```

Reply:
using global fetch doesn't work in the tests, axios will because it's isomorphic (works on client and server)

Also there maybe two error return types, AxiosError and another Error,

I'm not sure getOrElseW is being used correctly here as it should be an Either<good, bad> return type, and also the error message is a bit naff

Reply:
Done